src/libs/ajax/DatasetBuilder.ts

```diff
@@ -69,23 +95,88 @@ export interface GetConceptsResponse {
 type DatasetRequest = {
   cohorts: Cohort[];
   conceptSets: ConceptSet[];
-  valuesSets: { domain: string; values: DatasetBuilderValue[] }[];
+  valueSets: { domain: string; values: DatasetBuilderValue[] }[];
+};
+
+export type DatasetRequestAPI = {
+  cohorts: CohortAPI[];
+  conceptSets: ConceptSet[];
+  valueSets: { id: number; name: string; values: string[] }[];
 };
 
-type DatasetAccessRequest = {
+export type DatasetAccessRequest = {
   name: string;
   researchPurposeStatement: string;
   datasetRequest: DatasetRequest;
 };
 
+export type DatasetAccessRequestAPI = {
+  name: string;
+  researchPurposeStatement: string;
+  datasetRequest: DatasetRequestAPI;
+};
+
+export const convertValueSets = (valueSets: { domain: string; values: DatasetBuilderValue[] }[]) => {
+  return _.map(
+    (valueSet) => ({ id: 0, name: valueSet.domain, values: _.map(({ name }) => name, valueSets.values) }),
+    valueSets
+  );
+};
+
+export const convertCohorts = (cohorts: Cohort[]) => {
+  return _.map(
+    (cohort) => ({
+      name: cohort.name,
+      criteriaGroups: _.map(
+        (criteriaGroup) => ({
+          name: criteriaGroup.name,
+          mustMeet: criteriaGroup.mustMeet,
+          meetAll: criteriaGroup.meetAll,
+          count: criteriaGroup.count,
+          criteria: _.map((criteria) => convertCriteria(criteria), criteriaGroup.criteria),
+        }),
+        cohort.criteriaGroups
+      ),
+    }),
+    cohorts
+  );
+};
+
+export const convertCriteria = (criteria: AnyCriteria) => {
```

Comment:
Not sure what the convention in other places in terra-ui is, but I think it'd be nice to declare the return types for these conversion methods, so it's apparent to the reader what types are being returned. For example, I'd add the type here like this:
```suggestion
export const convertCriteria = (criteria: AnyCriteria): AnyCriteriaAPI => {
```